games/rpg/src/lib.rs

```diff
@@ -0,0 +1,1483 @@
+                    ability_scaling: item.ability_scaling, range: item.range, version: self.version,
+                };
+                self.system_vault.authorize(||item_bucket.burn());
+                let new_bucket = self.system_vault.authorize(||
+                    borrow_resource_manager!(self.item_nft)
+                        .mint_non_fungible(&NonFungibleId::random(), new));
+                return new_bucket
+            }
+            else if which == 2 {
+                let mut item: Armor = item_bucket.non_fungibles()[0].data();
+                let item2: Armor = item_bucket.non_fungibles()[1].data();   
+                assert!(item.level == item2.level);            
+                assert!(item.id == item2.id);
+                assert!(item.part == item2.part);
+                item.level += dec!(1); 
+                item.health *= dec!("1.2");
+                item.defense *= dec!("1.2");
+                item.ability_scaling *= dec!("1.2");  
+                let new = Armor { id: item.id, part: item.part, level: item.level, health: item.health,
+                    defense: item.defense, weight: item.weight, ability: item.ability, ability_scaling: item.ability_scaling, version: self.version,
+                    };
+                self.system_vault.authorize(||item_bucket.burn());
+                let new_bucket = self.system_vault.authorize(||
+                    borrow_resource_manager!(self.item_nft)
+                        .mint_non_fungible(&NonFungibleId::random(), new));
+                return new_bucket
+            }
+            else {
+                let mut item: Accessory = item_bucket.non_fungibles()[0].data();
+                let item2: Accessory = item_bucket.non_fungibles()[1].data();  
+                assert!(item.level == item2.level);        
+                assert!(item.id == item2.id);
+                assert!(item.part == item2.part);
+                item.level += dec!(1); 
+                item.attack *= dec!("1.2");
+                item.magic *= dec!("1.2");
+                item.speed *= dec!("1.2");
+                item.ability_scaling *= dec!("1.2");  
+                let new = Accessory { id: item.id, part: item.part, level: item.level, attack: item.attack,
+                    magic: item.magic, speed: item.speed, weight: item.weight, ability: item.ability, ability_scaling: item.ability_scaling, version: self.version,
+                    };
+                self.system_vault.authorize(||item_bucket.burn());
+                let new_bucket = self.system_vault.authorize(||
+                    borrow_resource_manager!(self.item_nft)
+                        .mint_non_fungible(&NonFungibleId::random(), new));
+                return new_bucket
+            }              
+        }
+        // Creates weapons
+        pub fn create_weapon_1(&mut self, gold: Bucket, resource1: Bucket, resource2: Bucket, class: u8) -> Bucket {
```

Comment:
I think that the `class` can be changed from a `u8` to be an enum to help make the code clearer.